servers/rendering/renderer_rd/renderer_canvas_render_rd.cpp

```diff
@@ -2146,23 +2146,26 @@ void RendererCanvasRenderRD::_render_batch_items(RenderTarget p_to_render_target
 				current_batch->material_data = material_data;
 			}
 
-			Transform2D base_transform = p_canvas_transform_inverse * ci->final_transform;
 			if (!ci->repeat_size.x && !ci->repeat_size.y) {
```

Comment:
I think non-zero `repeat_size` already guarantees non-null `repeat_source_item`... but probably safer to ensure this is indeed the case in the `else` block? :thinking:
```suggestion
			if (!ci->repeat_source_item || (!ci->repeat_size.x && !ci->repeat_size.y)) {
```

Reply:
@kleonc yes, according to this, it is, so we can change the logic:

https://github.com/godotengine/godot/blob/1bd8372813d8a329188c05e8cc8c0c66f60b1735/servers/rendering/renderer_canvas_cull.cpp#L462-L472

Reply:
Same with when it is set in mirroring:

https://github.com/godotengine/godot/blob/1bd8372813d8a329188c05e8cc8c0c66f60b1735/servers/rendering/renderer_canvas_cull.cpp#L446-L460

Reply:
Yeah, makes more sense to check it here and remove `if (ci->repeat_source_item)` check from the inner loop in the `else` block.

Reply:
Done